BuildingAndTestingInBunsen.md

```diff
@@ -82,38 +85,76 @@ Assuming the repository was cloned to your homedir:
     make
     ```
 
-# Testing
+# Automated Testing
+
+## make jenkins
 
 As part of the checkin process, VDO's projects always undergo testing via Jenkins.  This testing is initiated by calling the `make jenkins` target.
 
 The `make jenkins` target is implemented in both the [common](https://github.com/dm-vdo/common) and [vdo-devel](https://github.com/dm-vdo/vdo-devel) projects.
 
-IMPORTANT: It all contributions made to these projects will run through the `make jenkins` target, so it is a good idea to run this up front to avoid any surprises.  It is generally the contributor's responsibility to ensure that all tests pass.
+IMPORTANT: All contributions made to these projects will run through the `make jenkins` target, so it is a good idea to run this before submitting a pull request to avoid any surprises.  It is generally the contributor's responsibility to ensure that all tests pass.
+
+## make checkin
+
+A quicker, but less complete, alternative to `make jenkins` in the repositories is the `make checkin` target which will run a less extensive suite of tests set of tests against UDS and VDO.
 
----
+# Testing with Perl
+
+## Running specific perl tests
 
 If you want to run specific tests, then you should find the relevant directory to start:
-* vdo-devel - src/perl/vdotest/
+* vdo-devel:
+  * VDO tests - src/perl/vdotest/
+  * UDS tests - src/c++/uds/src/tests
+    * These tests are built and run in a kernel module via the perl test infrastructure.
+    * Test files can be identified by their _t{1..5}, _n{1..5}, p{1,2}, x1 suffixes.
```

Comment:
Could describe it, as I think the code does, as a regex. MAyebe, as "Test names match "^\w*_[tnpx]\d+$" -- that is, a name, followed by '_', one of the characters 'tnpx', and a digit." Or shrug. The specificity on the digits seems unnecessary is all I"m really trying to say.

Reply:
Addressed in latest update.